### ISSUE TITLE
Change the accent color of Scratch Addons

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#6027]
### The problem

Currently, Scratch Addons' accent color is `#ff7b26`, under the CSS variable ["--brand-orange"](https://github.com/ScratchAddons/ScratchAddons/blob/3b4fcf6d1003fd78a5902f652b928696cc5ba915/webpages/styles/colors.css#L2). However, there are a number of places where this isn't ideal. Here are some examples in light mode following the WCAG:

| Colors | Contrast Ratio | AA requirement | AAA requirement |
|--------|--------|--------|--------|
| Scratch Addons logo and name on brand orange | 2.59 | 3 | 4.5 |
| Small white text on brand orange | 2.59 | 4.5 | 7 |
| Switches set to on and selected single-select settings on content background | 2.42 | 3 | 4.5 |

There are probably some other locations where this is a problem, too. That's why I'm bringing up the suggestion to change the accent color, as big a change as it may sound.

I'd like to hear your thoughts and ideas. We should work together to find a better color to use as Scratch starts to do the same thing. I suggest making this change non-optional and balancing accessibility and design. We can measure how acceptable this change will be long-term by putting ourselves in the shoes of someone who has just started to use Scratch Addons after this change - they will not have to get used to the change.

I originally suggested considering this change in a [discussion](https://github.com/ScratchAddons/ScratchAddons/discussions/5916), but I think an issue will be seen by more contributors.

**Comment by Secret-chest** [comment]
> We can measure how acceptable this change will be long-term by putting ourselves in the shoes of someone who has just started to use Scratch Addons after this change - they will not have to get used to the change.

So basically you're going to disregard existing users? If Scratch changes to purple, maybe green could do. This would go well with customisable SA themes.

**Comment by WorldLanguages** [comment]
Reposting my comment from the discussion: 

> Personally regarding the brand color (which is also used in the logo), I'm open to changing it to a completely different color. In other words, I don't think it necessarily needs to be a shade of orange, so we shouldn't restrict our brand color to that. It would be a pretty drastic change for our brand and users would notice it, but I think we could try to think outside the box, at least for a bit.

**Comment by WorldLanguages** [comment]
What word comes to mind when thinking about the word "addons"? Personally, dark-ish green :P

**Comment by DNin01** [comment]
How about a mint green to sky blue gradient color? Depending on the background and text color, we could use the following linear gradients:
linear-gradient(135deg, `#008a49`, `#0062a8`)
linear-gradient(135deg, `#1af28c`, `#4bb4ff`)

If this sounds like a plan, I think I can do the rest (I'll see if I can make a PR to do several color changes).

**Comment by DNin01** [comment]
[Work has begun.](https://github.com/DNin01/ScratchAddons/tree/use-higher-contrast-colors)

**Comment by WorldLanguages** [comment]
The accent color could theoretically be a gradient color, but if we want the logo to match the brand color, it becomes complicated. It's hard to use gradients in the extension bar as there's very few screen pixels to play with.

**Comment by Secret-chest** [comment]
Do gradients fit Scratch? It should be mint green

**Comment by Secret-chest** [comment]
Why are setting page customisations rejected?

**Comment by Secret-chest** [comment]
Should we do this along the big redesign (#5893)?

**Comment by WorldLanguages** [comment]
> Do gradients fit Scratch? It should be mint green

Our accent color is rarely used inside the Scratch website, so it doesn't really matter.

> Should we do this along the big redesign (#5893)?

Not necessarily.

**Comment by Samq64** [comment]
Kind of unrelated, but even though I was the one who changed the colour in pre-release versions, I think there are better solutions. Maybe a badge next to the extension name like the website?

**Comment by Secret-chest** [comment]
We're making some forced changes in an extension designed to be customisable.

**Comment by WorldLanguages** [comment]
> Kind of unrelated, but even though I was the one who changed the colour in pre-release versions, I think there are better solutions. Maybe a badge next to the extension name like the website?

Could you clarify? Are you suggesting to use the prerelease blue in the stable version that reaches users, and find another way to indicate prerelease?

> We're making some forced changes in an extension designed to be customisable.

Not sure if "forced" is the right word when you can change it back by downloading a zip and opening some files using notepad.

**Comment by Samq64** [comment]
I guess we could add an option to change the colour in more settings, but it would be tricky to make it work outside the popup and settings pages.

**Comment by DNin01** [comment]
> We're making some forced changes in an extension designed to be customisable.

Yeah, I was worried about that.

**Comment by Secret-chest** [comment]
> > We're making some forced changes in an extension designed to be customisable.
> 
> Yeah, I was worried about that.

So, we'll add settings?

**Comment by DNin01** [comment]
> > > We're making some forced changes in an extension designed to be customisable.
> > 
> > 
> > Yeah, I was worried about that.
> 
> So, we'll add settings?

We could try to allow returning the classic theme, but it would mean more to maintain and potentially more complicated CSS, all for an issue that users will eventually get over. But I understand why users might not like some much more noticeable changes, such as changing the brand/accent color. I tried to still make it look pretty enough to be usable while ensuring the contrast was okay. If this is a huge deal, we could potentially allow reverting some more major changes but not others to save time.

**Comment by apple502j** [comment]
Note: WCAG guidelines are known to have some issues. We shouldn't automatically determine against one set of rules, especially when brand color is at play.

**Comment by Secret-chest** [comment]
Gradients are never pretty enough to me. Scratch Addons now looks like that Romanian shopping website.

**Comment by WorldLanguages** [comment]
Worth reading (really): https://www.bounteous.com/insights/2019/03/22/orange-you-accessible-mini-case-study-color-ratio

**Comment by DNin01** [comment]
> The data set of users preferred the white text button over the black text button, primarily because of contrast. But different types of color blindness produced different results. Specifically, the monochrome/achromatopsia user preferred black text. I also uncovered some interesting legibility concerns. Some users had unique issues reading the black text, saying it caused buzzing and headaches. Even though the black text is the accessible option by WCAG standards, it fails to account for this particular level of accessibility.

I can see that.

**Comment by WorldLanguages** [comment]
My point is that it will be hard to generate consensus to change the accent/brand color if WCAG color guidelines are the main justification.
I believe there are some other reasons we may want to change the brand color, like more clearly distinguishing our brand from Scratch's, which also uses orange in some places. We may also want to change the font family and accent color at the same time, as part of a general revamp of the settings page and other parts of the extension. There might be other reasons, so this is why I have tried not to strongly oppose suggestions like this.

**Comment by BroJac5246** [comment]
I think if we do something like this then it should be when SA 2.0 is released, since it'll look less out-of-place.

**Comment by apple502j** [comment]
Also, remember that brand color is mostly used in places where color accessibility matters the least - banners and icons. If this was text color or background this would've had a higher priority.

**Comment by DNin01** [comment]
Let's do this change later. I've still got the rest of the changes I made ready for review.

**Comment by mybearworld** [comment]
> Also, remember that brand color is mostly used in places where color accessibility matters the least - banners and icons. If this was text color or background this would've had a higher priority.

It *is* used for setting icons.

**Comment by DNin01** [comment]
> > Also, remember that brand color is mostly used in places where color accessibility matters the least - banners and icons. If this was text color or background this would've had a higher priority.
> 
> 
> 
> It *is* used for setting icons.

Like the ones used in Scratch Notifier?

**Comment by mybearworld** [comment]
> > > Also, remember that brand color is mostly used in places where color accessibility matters the least - banners and icons. If this was text color or background this would've had a higher priority.
> > 
> > 
> > It _is_ used for setting icons.
> 
> Like the ones used in Scratch Notifier?

Yes, these:
![Setting icons](https://github.com/ScratchAddons/ScratchAddons/assets/130385691/9faf4738-c111-468c-af06-385f5c24c336)

**Comment by Secret-chest** [comment]
> It _is_ used for setting icons.

Sure, but they're not important: there is a text.
